src/commonMain/kotlin/character/classes/warrior/abilities/Execute.kt

```diff
@@ -40,11 +40,13 @@ class Execute : Ability() {
 
     override fun cast(sp: SimParticipant) {
         val item = sp.character.gear.mainHand
-        val damage = 925.0 + (sp.getResource(resourceType(sp))).currentAmount * 21
+        val res = sp.resources[resourceType(sp)]
+        if(res == null) return
```

Comment:
Here I don't know if it should just fail silently like this, it's effectively impossible anyways since `Rotation.next()` already makes sure the resource returned by `resourceType(sp)` exists. Still, if it was possible, how do you want that handled?

Reply:
In cases where it should never fail (and deserves a crash when it does), we can just force the compiler to consider it non-nullable with `!!`